src/Tes.Runner/Authentication/CredentialsManager.cs

```diff
@@ -37,7 +38,14 @@ private TimeSpan SleepDurationHandler(int attempt)
 
         public TokenCredential GetTokenCredential(RuntimeOptions runtimeOptions)
         {
-            return retryPolicy.Execute(() => GetTokenCredentialImpl(runtimeOptions));
+            try
+            {
+                return retryPolicy.Execute(() => GetTokenCredentialImpl(runtimeOptions));
+            }
+            catch (CredentialUnavailableException)
```

Comment:
This will never happen here. It happens in `HttpApiClient` only. Also, it may be a `AuthenticationFailedException` (and not necessarily `CredentialUnavailableException`).